src/actions/preferences.js

```diff
@@ -95,29 +82,18 @@ export function makeGroupMessageVisibleIfNecessary(channelId) {
 }
 
 export function savePreferences(preferences) {
-    return async (dispatch, getState) => {
-        dispatch({type: PreferencesTypes.SAVE_PREFERENCES_REQUEST}, getState);
-
-        try {
-            await Client.savePreferences(preferences);
-        } catch (error) {
-            forceLogoutIfNecessary(error, dispatch);
-            dispatch(batchActions([
-                {type: PreferencesTypes.SAVE_PREFERENCES_FAILURE, error},
-                getLogErrorAction(error)
-            ]), getState);
-            return;
-        }
-
-        dispatch(batchActions([
-            {
-                type: PreferencesTypes.RECEIVED_PREFERENCES,
-                data: preferences
-            },
-            {
-                type: PreferencesTypes.SAVE_PREFERENCES_SUCCESS
+    return async (dispatch) => {
+        dispatch({
+            type: PreferencesTypes.SAVE_PREFERENCES,
+            data: preferences,
+            meta: {
+                offline: {
+                    effect: () => Client.savePreferences(preferences),
+                    commit: {
+                        type: PreferencesTypes.RECEIVED_PREFERENCES
+                    }
```

Comment:
I can add one in. The rollback in this case won't run for an hour since it'll be ran against the default decaying schedule.